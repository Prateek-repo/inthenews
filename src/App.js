import './App.css'
import React, { Component } from 'react'
import NavBar from './components/navbar/NavBar'
import News from './components/news/News'
import {
  BrowserRouter as Router,
 
  Route,
  Routes
} from "react-router-dom";
import LoadingBar from 'react-top-loading-bar'

export default class App extends Component {

  state={
    progress: 0,
    country: "in",
    darkModeStatus: false
  }


  setProgress = (theBarProgress) => {
    this.setState({
      progress: theBarProgress
    })
  }

  setCountry = (country) => {
    console.log("dfd", country)
    this.setState({country: country})
  }

  darkModeEnable = () => {
    this.setState({darkModeStatus: !this.state.darkModeStatus})
      }

  routeElement = (category, key) => {
    const { country, darkModeStatus} = this.state
    return (<News progressBar={this.setProgress} key={key === "/" ? "general" : key} pageSize={6} country={country} category={category === "/" ? "general" : category} darkMode={darkModeStatus}/>)
  }

  render() {
    const {progress, category, darkModeStatus} = this.state
    return (
      <div style={{backgroundColor: darkModeStatus ? "black" : null}}>
        <Router>
        <NavBar setTheCountry={this.setCountry} category={category} darkMode={this.darkModeEnable}/>
        <LoadingBar
        color='#f11946'
        progress={progress}
        shadow={true}
        height={3}
        onLoaderFinished={() => this.setProgress(0)}
      />
          <Routes>
          <Route exact path="/" element={this.routeElement("/", "/")}/>
          <Route exact path="/general" element={this.routeElement("general", "general")}/>
          <Route exact path="/health" element={this.routeElement("health", "health")}/>
          <Route exact path="/business" element={this.routeElement("business", "business")}/>
          <Route exact path="/entertainment" element={this.routeElement("entertainment", "entertainment")}/>
          <Route exact path="/sports" element={this.routeElement("sports", "sports")}/>
          <Route exact path="/technology" element={this.routeElement("technology", "technology")}/>
          <Route exact path="/science" element={this.routeElement("science", "science")}/>
          </Routes>
       
       
        
       </Router>
      </div>
    )
  }
}


