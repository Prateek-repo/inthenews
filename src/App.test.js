import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import LoadingBar from "react-top-loading-bar";

import App from "./App";

configure({ adapter: new Adapter() });

describe("<App/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it("SnapShot creation", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("handling setProgress", () => {
    const componentInstance = wrapper.instance();
    componentInstance.setProgress(10);
    expect(componentInstance.state.progress).toEqual(10);
  });

  it("handling setCountry", () => {
    const componentInstance = wrapper.instance();
    componentInstance.setCountry("in");
    expect(componentInstance.state.country).toEqual("in");
  });

  it("handling darkModeEnable", () => {
    wrapper.setState({
      darkModeStatus: true,
    });
    const componentInstance = wrapper.instance();
    componentInstance.darkModeEnable();
    expect(componentInstance.state.darkModeStatus).toEqual(false);
  });

  it("should be handling LoadingBar", () => {
    wrapper.find(LoadingBar).props().onLoaderFinished();
    const componentInstance = wrapper.instance();
    expect(componentInstance.state.progress).toEqual(0);
  });
});
