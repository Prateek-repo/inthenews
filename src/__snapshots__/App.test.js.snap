// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> SnapShot creation 1`] = `
<div
  style={
    Object {
      "backgroundColor": null,
    }
  }
>
  <BrowserRouter>
    <NavBar
      darkMode={[Function]}
      setTheCountry={[Function]}
    />
    <ForwardRef
      color="#f11946"
      height={3}
      onLoaderFinished={[Function]}
      progress={0}
      shadow={true}
    />
    <Routes>
      <Route
        element={
          <News
            category="general"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/"
      />
      <Route
        element={
          <News
            category="general"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/general"
      />
      <Route
        element={
          <News
            category="health"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/health"
      />
      <Route
        element={
          <News
            category="business"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/business"
      />
      <Route
        element={
          <News
            category="entertainment"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/entertainment"
      />
      <Route
        element={
          <News
            category="sports"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/sports"
      />
      <Route
        element={
          <News
            category="technology"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/technology"
      />
      <Route
        element={
          <News
            category="science"
            country="in"
            darkMode={false}
            pageSize={20}
            progressBar={[Function]}
          />
        }
        exact={true}
        path="/science"
      />
    </Routes>
  </BrowserRouter>
</div>
`;
